tests/common/helpers/pfc_storm.py

```diff
@@ -0,0 +1,258 @@
+        self.fanout_hosts = fanouthosts
+        self.pfc_gen_file = kwargs.pop('pfc_gen_file', "pfc_gen.py")
+        self.pfc_queue_idx = kwargs.pop('pfc_queue_index', 3)
+        self.pfc_frames_number = kwargs.pop('pfc_frames_number', 100000)
+        self.peer_info = kwargs.pop('peer_info')
+        self._validate_params(expected_args=['pfc_fanout_interface', 'peerdevice'])
+        if 'hwsku' not in self.peer_info:
+            self._populate_peer_hwsku()
+        self.platform_name = None
+        self.update_platform_name()
+        self._populate_optional_params(kwargs)
+        self.peer_device = self.fanout_hosts[self.peer_info['peerdevice']]
+
+    def _populate_peer_hwsku(self):
+        """
+        Find out the hwsku associated with the fanout
+        """
+        peer_dev_info = self.fanout_info[self.peer_info['peerdevice']]['device_info']
+        self.peer_info['hwsku'] = peer_dev_info['HwSku']
+
+    def _validate_params(self, **params):
+        """
+        Validate if all the needed keys are present
+        """
+        expected_args = params.get('expected_args')
+        peer_info_keys = self.peer_info.keys()
+        if not all(elem in peer_info_keys for elem in expected_args):
+            raise MissingInputError("Peer_info does not contain all the keys,"
+                                    "Expected args: {}".format(expected_args))
+
+    def _populate_optional_params(self, kwargs):
+        """
+        Create var and assign values if any the following keys are present
+        'pfc_storm_defer_time', 'pfc_storm_stop_defer_time', 'pfc_asym'
+        """
+        if len(kwargs) > 0:
+            for key in kwargs:
+                setattr(self, key, kwargs.get(key))
+        kwargs.clear()
+
+    def _create_pfc_gen(self):
+        """
+        Create the pfc generation file on the fanout if it does not exist
+        """
+        out = self.peer_device.stat(path="/mnt/flash")
+        if not out['stat']['exists'] or not out['stat']['isdir']:
+            self.peer_device.file(path="/mnt/flash/{}".format(self.pfc_gen_file), state="touch")
+
+    def deploy_pfc_gen(self):
+        """
+        Deploy the pfc generation file on the fanout
+        """
+        if 'arista' in self.peer_info['hwsku'].lower():
+            self._create_pfc_gen()
+            self.peer_device.copy(src="common/helpers/{}".format(self.pfc_gen_file),
+                                  dest="/mnt/flash")
+
+    def update_queue_index(self, q_idx):
+        """
+        Update the queue index. Can be invoked after the class init to change the queue index
+        """
+        self.pfc_queue_idx = q_idx
+
+    def update_peer_info(self, peer_info):
+        """
+        Update the fanout info. Can be invoked after the class init to change the fanout or fanout interface
+        """
+        self._validate_params(expected_args=['peerdevice', 'pfc_fanout_interface'])
+        for key in peer_info:
+            self.peer_info[key] = peer_info[key]
```

Comment:
update()?

Reply:
change the usage to use dict update? subscript option is slightly faster than update though shouldn't matter much here